geospace-lib/src/main/scala/com/socrata/geospace/lib/shapefile/ZipUtils.scala

```diff
@@ -73,6 +78,55 @@ class TemporaryZip(compressed: Array[Byte]) extends Closeable with Logging {
     FileUtils.deleteDirectory(contents)
     logger.info("Deleted temporary extracted shapefile contents")
     archive.delete()
+    tmpDir.foreach(_.delete)
     logger.info("Deleted temporary shapefile zip")
   }
+
+}
+
+
+/** Saves the contents of a zip file in a single flattened directory
+  * and cleans up all temporary files afterwards.
+  *
+  * @constructor create a new temporary zip given the zip file as a byte array.
+  * @param compressed byte array representation of the zip file
+  */
+case class ZipFromArray(compressed: Array[Byte]) extends AbstractZip(None) {
+  require(Option(compressed).filter(_.nonEmpty).isDefined, "Null or empty zip file")
+
+  def getArchive = {
+    for {
+      in <- managed(new ByteArrayInputStream(compressed))
+      out <- managed(new FileOutputStream(tmpFile))
+    } {
+      IOUtils.copy(in, out)
+      logger.info("Temporarily copied shapefile zip to {}", tmpFile.getAbsolutePath)
+    }
+
+    tmpFile
+  }
 }
+
+/** Saves the contents of a zip file in a single flattened directory
+  * and cleans up all temporary files afterwards.
+  *
+  * @constructor create a new temporary zip given the zip file inputstream.
+  * @param compressed inputstream representation of the zip file
+  * @param tmpDir temp directory where to place the zip file.
+  */
+case class ZipFromStream(compressed: InputStream, tmpDir: Option[File]) extends AbstractZip(tmpDir) {
+  require(Option(compressed).filter(_.available >= 0).isDefined, "Null or empty zip file")
+
+  def getArchive = {
+    val ops = new FileOutputStream(tmpFile)
+
+    try {
+      IOUtils.copy(compressed, ops)
```

Comment:
Should we use a managed() block like the `ZipFromArray` above, just for consistency?  If not, please comment.